python/heterocl/schedule.py

```diff
@@ -292,66 +286,298 @@ def to(self, tensors, dst, src=None, axis=0,
+                if isinstance(target, tuple):
+                    _, target = target
+
             # convert hcl stage
-            try: dst = self[dst]
+            try: 
+                if isinstance(dst, tuple):
+                   dst, _ = dst 
+                dst = self.__getitem__(dst)
             except: pass
 
             move_to_device = False
             if src is None:
                 # move to device
-                if isinstance(dst, Device) or \
-                        isinstance(dst, DevMediaPair):
+                # Example: s.to(A, target.FPGA)
+                if isinstance(dst, Device) or isinstance(dst, DevMediaPair):
                     if axis == 0:
                         move_to_device = True
                     else: # inner-stage movement
                         assert isinstance(tensor, Stage)
-                        target = self[tensor]
-
-                else: # inter-stage
-                    src = self[tensor]
+                        target = self.__getitem__(tensor)
+                    # burst_len == -1 means burst is diabled 
+                    if burst == True:
+                        if burst_len < 0: burst_len = 0
+                    else:
+                        assert burst_len == -1, "The burst mode must be True " + \
+                            "before setting the burst length..."
+                # inter-stage
+                # Example: s.to(A, stage) where the stage consumes tensor A
+                # in this case, the stage producing the tensor A is src stage
+                else: 
+                    src = self.__getitem__(tensor)
+
+            # inter-stage data movement
+            if not (isinstance(dst, Device) or isinstance(dst, DevMediaPair)):
+                # 1. handle inter-kernel data streaming 
+                if isinstance(src.op, _tensor.PlaceholderOp) and isinstance(dst.op, _tensor.PlaceholderOp): 
+                    # search the kernel calls globally and find the target tensor
+                    print("[ INFO ] inter-kernel streaming for target tensor")
+                    src_stage_name = src.op.name.split(".")[1]
+                    dst_stage_name = dst.op.name.split(".")[1]
+                    assert src_stage_name in self.mod_calls
+                    assert dst_stage_name in self.mod_calls
+                    if len(self.mod_calls[src_stage_name]) > 1:
+                        raise HCLError("{} has more than one call sites".format(src_stage_name))
+                    if len(self.mod_calls[dst_stage_name]) > 1:
+                        raise HCLError("{} has more than one call sites".format(dst_stage_name))
+                    
+                    def get_overlap(a, b):
+                        return list(set(a) & set(b))[0]
+
+                    target = get_overlap(self.mod_calls[dst_stage_name][0],
+                        self.mod_calls[src_stage_name][0])
+                    target = target.tensor
```

Comment:
I just found that `target` is not defined before. The name is confusing with `dst`. Those two sound like the same thing.

Reply:
What is `target` by the way?

Reply:
target is the target tensor, while destination is the destination stage. Oftentimes the target tensor (to be moved) is not produced by the destination stage, so I created these arguments.

Reply:
Do we want to rename it?

Reply:
Shouldn't the target tensor be `tensor`? `target` is used but not defined in the code, which is unsafe.

Reply:
Actually we have some preprocessing on the input `tensor`.